client/helpers.go

```diff
@@ -382,3 +391,28 @@ func TagsToMap(tagSlice interface{}) map[string]string {
 	TagsIntoMap(tagSlice, ret)
 	return ret
 }
+
+func ListAndDetailResolver(ctx context.Context, meta schema.ClientMeta, res chan<- interface{}, list ListResolver, details DetailResolver) error {
+	sem := semaphore.NewWeighted(int64(MAX_GOROUTINES))
+	errs, ctx := errgroup.WithContext(ctx)
+	response, err := list(ctx, meta)
+	if err != nil {
+		return diag.WrapError(err)
+	}
+	for _, item := range response {
+		if err := sem.Acquire(ctx, 1); err != nil {
+			return diag.WrapError(err)
+		}
+		func(summary interface{}) {
+			errs.Go(func() error {
+				defer sem.Release(1)
+				return details(ctx, meta, res, summary)
```

Comment:
if one details fails they all fail as its an error group, perhaps aggregate all diags and return in the end?

Reply:
Good suggestion